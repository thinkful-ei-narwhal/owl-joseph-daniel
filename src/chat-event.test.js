import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatEvent from './chat-event';


describe('<ChatEvent />', () => {
  it('works with message', () => {
    const div = document.createElement('div')
    ReactDOM.render(<ChatEvent id={1} type={'message'} message={'Hello world'} time={1548852646559} timestamp={1548852484247} name={'Bob'}/>, div);
    // if something is wrong, itd fail here
    ReactDOM.unmountComponentAtNode(div);
  });

  it('works without message', () => {
    const div = document.createElement('div')
    ReactDOM.render(<ChatEvent id={1} type={'thumbs-down'} timestamp={1548852484247} name={'Bob'}/>, div);
    // if something is wrong, itd fail here
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders this UI as expected', () => {
    const tree = renderer.create(<ChatEvent id={1} type={'message'} message={'Hello world'} time={1548852646559} timestamp={1548852484247} name={'Bob'}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});