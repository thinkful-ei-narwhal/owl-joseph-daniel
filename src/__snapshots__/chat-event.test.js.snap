// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatEvent /> renders this UI as expected 1`] = `
<div
  className="Chat-event"
>
  <img
    alt="Koren Templeton"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <h3>
    Koren Templeton
  </h3>
  <p>
    7:50
  </p>
  <p>
    Hello world
  </p>
  <p>
    
  </p>
</div>
`;
